### ISSUE TITLE
[Feature Request]: Add Function `NaiveSyncBatchNorm1d` to ivy backend and frontend 

### ISSUE THREAD

**Issue by rajveer43** [unifyai/ivy#26949]
**Is your feature request related to a problem? Please describe.**
This feature request  `NaiveSyncBatchNorm1d` is an extension of the existing `nn.BatchNorm1d` module, designed to support synchronization across multiple devices, either locally or globally. 

The motivation behind this feature request is to enhance Ivy's capabilities in distributed deep learning scenarios. the proposed `NaiveSyncBatchNorm1d` module would fill this gap.

I propose the addition of the `NaiveSyncBatchNorm1d` module to Ivy. The module provides the following features:
- Synchronization of batch normalization statistics (mean and variance) across multiple devices.
- Support for both global synchronization and local synchronization based on user requirements.

**Describe the solution you'd like**

**Describe alternatives you've considered**
While there are alternative ways to implement synchronized batch normalization, `NaiveSyncBatchNorm1d` offers a simple and efficient solution. 

**Additional context**
Here's an example of how `NaiveSyncBatchNorm1d` can be used:

```python
sync_bn = NaiveSyncBatchNorm1d(num_sync_devices=4, global_sync=False, num_features=64)
output = sync_bn(input_tensor)
```

**Comment by rajveer43** [comment]
@Dsantra92 it will be a great addition.

**Comment by Dsantra92** [comment]
We are not really focusing on distributed learning at the moment, will take a look at the implementation in a few hours.

**Comment by rajveer43** [comment]
@Dsantra92 what do you think??
